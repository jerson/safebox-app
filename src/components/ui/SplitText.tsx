import React from 'react';
import { View, StyleSheet, ViewStyle, TextStyle } from 'react-native';
import Colors from '../../modules/constants/Colors';
import Text, { TextProps } from './Text';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center'
  } as ViewStyle,
  split: {
    height: 1,
    backgroundColor: Colors.primaryLight,
    opacity: 0.5,
    position: 'absolute',
    left: 0,
    right: 0,
    top: 10
  } as ViewStyle,
  label: {
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 4,
    paddingBottom: 4,
    minWidth: 20,
    textAlign: 'center',
    color: Colors.white,
    backgroundColor: Colors.primary,
    lineHeight: 14,
    fontSize: 13,
    borderRadius: 10
  } as TextStyle
});
export interface SplitTextProps extends TextProps {
  title?: string;
}

const SplitText: React.FC<SplitTextProps> = ({ title, style, ...props }) => {
  return (
    <View style={[styles.container, style]}>
      <View style={styles.split} />
      <Text style={styles.label}>{title}</Text>
    </View>
  );
};

export default SplitText;
