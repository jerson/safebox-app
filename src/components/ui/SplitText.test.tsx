import React from 'react';
import * as renderer from 'react-test-renderer';
import SplitText, { SplitTextProps } from './SplitText';

const props: SplitTextProps = {
  title: 'sample'
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<SplitText {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<SplitText {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
