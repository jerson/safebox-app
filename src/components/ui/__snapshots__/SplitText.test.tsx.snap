// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <_class
    style={
      Array [
        Object {
          "backgroundColor": "#9066f5",
          "height": 1,
          "left": 0,
          "opacity": 0.5,
          "position": "absolute",
          "right": 0,
          "top": 12,
        },
        false,
        false,
      ]
    }
  />
  <_class
    style={
      Array [
        Object {
          "backgroundColor": "#4531cd",
          "borderRadius": 10,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 6,
        },
        false,
        false,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito",
            "fontSize": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
          },
          Object {},
          Object {
            "color": "#FFFFFF",
            "fontSize": 13,
            "lineHeight": 14,
            "minWidth": 20,
            "textAlign": "center",
          },
        ]
      }
    >
      sample
    </Text>
  </_class>
</_class>
`;
